src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -269,10 +330,20 @@ std::string MutationReplaceBinaryOperator::GenerateMutatorFunction(
     mutant_offset++;
   }
 
-  new_function
-      << "  return " << arg1_evaluated << " "
-      << clang::BinaryOperator::getOpcodeStr(binary_operator_.getOpcode()).str()
-      << " " << arg2_evaluated << ";\n";
+  const clang::BuiltinType* exprType =
+      binary_operator_.getType()->getAs<clang::BuiltinType>();
+  if (exprType->isBooleanType() ||
+      (exprType->isInteger() && !exprType->isBooleanType() &&
+       !binary_operator_.isLValue())) {
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return !(" << getExpr(ast_context)
+                 << ");\n";
+    mutant_offset++;
+  }
```

Comment:
This handles inserting the `!` unary operator. Should any of the other unary operators be considered for insertion?

Reply:
During refactoring I have added the `~` operator to also be inserted.